packager/packager.cc

```diff
@@ -1745,24 +1754,42 @@ class ComputePackageSCCs {
                 this->stack.pop_back();
                 this->nodeMap[poppedPkgName].onStack = false;
                 auto &poppedPkgInfo = PackageInfoImpl::from(*(gs.packageDB().getPackageInfoNonConst(poppedPkgName)));
-                poppedPkgInfo.sccID_ = sccId;
+
+                switch (edgeType) {
+                    case core::packages::ImportType::Normal: {
+                        poppedPkgInfo.sccID_ = sccId;
+                        break;
+                    }
+                    case core::packages::ImportType::Test: {
+                        poppedPkgInfo.testSccID_ = sccId;
+                        break;
+                    }
+                }
```

Comment:
Can we make it so that we can use a pointer-to-member here, since the SCC we're modifying is constant over the entire `tarjan` run?  I don't have super-strong opinions on the right way to pass that in at the top-level, but I think the right direction is somehow to link the `edgeType` to the pointer-to-member that we want?  Maybe a method on `PackageInfo` that does that?

That might be Too Much C++, but my sense is that saving a branch here is valuable.

Reply:
Yeah, I think that's a good idea. I'll have a look at how we might do that at the `tarjan` call site.